### ISSUE TITLE
Nested smart components don't get variated 

### ISSUE THREAD

**Issue by simoncozens** [googlefonts/glyphsLib#880]
Noto Sans Math had `_part.braceright` (a smart glyph) which was itself implemented as a horizontally flipped component of `_part.braceleft` (another smart glyph). The nested component was not being variated, but copied directly into users of `_part.braceright`. I had to ensure that all glyphs starting with `_part.` had their components decomposed before it would work.

**Comment by anthrotype** [comment]
so IIUC, it's not a smart components that uses another smart components here, but rather a smart components that contains a regular (flipped) component?
Right now, it seems like smart_components.py is assuming that the smart component glyphs only contains paths (doesn't even look if there are any components).
Is it even possible to have a _smart_ component inside another _smart_ component, and what does that look like?

**Comment by simoncozens** [comment]
> it's not a smart components that uses another smart components here, but rather a smart components that contains a regular (flipped) component?

Not exactly. There *is* nested interpolation going on here. `_part.braceright` is a smart component with two layers: the first layer (with smart component mapping Size=1) contains a flipped version of `_part.braceleft` (a smart component) with smart component values Size=1 and the second layer (with smart component mapping Size=12) contains a flipped version of `_part.braceleft` with smart component values Size=12. From this, we derive `braceright.size1`, `braceright.size2` etc.
